File: UIKit and Animations/WWDC 2014/Building Adaptive Apps with UIKit/AdaptivePhotos An Adaptive Application/AdaptivePhotosAnAdaptiveApplication/AdaptivePhotos/AAPLUser.m
/*
 Copyright (C) 2014 Apple Inc. All Rights Reserved.
 See LICENSE.txt for this sample’s licensing information
 
 Abstract:
 
  The top level model object.
  
 */

#import "AAPLUser.h"
#import "AAPLConversation.h"
#import "AAPLPhoto.h"

@implementation AAPLUser

+ (instancetype)userWithDictionary:(NSDictionary *)dictionary
{
    AAPLUser *user = [[self alloc] init];
    user.name = [dictionary objectForKey:@"name"];
    
    NSArray *conversationDictionaries = [dictionary objectForKey:@"conversations"];
    NSMutableArray *conversations = [NSMutableArray array];
    
    for (NSDictionary *conversationDictionary in conversationDictionaries) {
        AAPLConversation *conversation = [AAPLConversation conversationWithDictionary:conversationDictionary];
        [conversations addObject:conversation];
    }
    
    user.conversations = conversations;
    
    NSDictionary *lastPhotoDictionary = [dictionary objectForKey:@"lastPhoto"];
    user.lastPhoto = [AAPLPhoto photoWithDictionary:lastPhotoDictionary];
    return user;
}

@end

File: UIKit and Animations/WWDC 2014/Building Adaptive Apps with UIKit/AdaptivePhotos An Adaptive Application/AdaptivePhotos_Swift/AdaptivePhotos_Swift/Model/AAPLUser.swift
/*
	Copyright (C) 2014 Apple Inc. All Rights Reserved.
See LICENSE.txt for this sample’s licensing information

Abstract:

The top level model object.

*/


import Foundation


class AAPLUser: NSObject {
	
	var name: String!
	var conversations: [AAPLConversation]!
	var lastPhoto: AAPLPhoto!
	
	static func userWithDictionary(dictionary: [String: Any]) -> AAPLUser {
		let user = AAPLUser()
		user.name = dictionary["name"] as? String
		let conversationDictionaries = dictionary["conversations"] as! [[String : Any]]
		var conversations: [AAPLConversation] = []
		
		for conversationDictionary in conversationDictionaries {
			let conversation = AAPLConversation.conversationWithDictionary(conversationDictionary)
			conversations.append(conversation)
		}
		
		user.conversations = conversations
		
		let lastPhotoDictionary = dictionary["lastPhoto"] as! [String : Any]
		user.lastPhoto = AAPLPhoto.photoWithDictionary(lastPhotoDictionary)
		return user
 	}
}
